tests/raftstore/transport_simulate.rs

```diff
@@ -265,3 +266,30 @@ impl FilterFactory for IsolateRegionStore {
              }]
     }
 }
+
+struct SnapshotFilter {
+    drop: AtomicBool,
```

Comment:
I see most of the filters use the same "drop" logic, set "drop" in before and then check and return error in after. 
Can you abstract it later, and let Filter more easy to implement.